Main.hs

```diff
@@ -4,22 +4,47 @@ import Control.Monad (unless)
 import Text.Printf (printf)
 
 short :: [a] -> Bool
-short = undefined 
+short (_:_:_:_) = False
+short _         = True
 
 lovely :: [Int] -> Bool
-lovely = undefined 
+lovely (_:_:14:_) = True
+lovely xs         = short xs
 
 rightTriangles :: [(Int, Int, Int)]
-rightTriangles = undefined 
+rightTriangles = [(a, b, c) | c <- [1..], a <- reverse [1..c], b <- [a..c], a^2 + b^2 == c^2]
```

Comment:
What is the point of making excessive reverse call?

Reply:
try swapping a and b in comprehension)

Reply:
[Fixed](https://github.com/kajigor/fp-2024-cub-fall/pull/1/commits/c0f2c95b320b069855d4d973b7dbc59d09824b86#diff-894c9c531eb0d7c5f20f08d4008d2dfe4e48d3e542af618ed58a9b281cd28aa3), thank you!